plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseJdbcConnectorTest.java

```diff
@@ -1090,6 +1091,34 @@ public void testArithmeticPredicatePushdown()
         // TODO add coverage for other arithmetic pushdowns https://github.com/trinodb/trino/issues/14808
     }
 
+    @Test
+    public void testComplexExpressionOnApproximateNumericPredicatePushdown()
+    {
+        if (!hasBehavior(SUPPORTS_PREDICATE_EXPRESSION_PUSHDOWN)) {
+            assertThat(query("SELECT orderkey FROM orders WHERE totalprice > 10 OR shippriority = 0"))
+                    .isNotFullyPushedDown(FilterNode.class);
+            return;
+        }
+        assertThat(query("SELECT orderkey FROM orders WHERE totalprice > 10 OR shippriority = 0")).isFullyPushedDown();
+
+        try (TestTable testTable = new TestTable(
+                getQueryRunner()::execute,
+                "complex_expression_",
+                "AS SELECT 1 as a_int, REAL '3.14' as a_real, DOUBLE '1.9891e30' as a_double")) {
+            assertThat(query("SELECT a_int FROM " + testTable.getName() + " WHERE a_real > REAL '3.12' OR a_double >  DOUBLE '1.9890e30'"))
+                    .isFullyPushedDown();
```

Comment:
can we add testcase with at least either  Double.MIN_VALUE or  Double.MIN_VALUE and Float.MIN_VALUE or  Float.MIN_VALUE

Reply:
good point

> with at least either Double.MIN_VALUE or Double.MIN_VALUE

not "or". all these should be tested